components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -2381,6 +2419,33 @@ private String doUserAuthorization(OAuthMessage oAuthMessage, String sessionData
         }
     }
 
+    /**
+     * Validate scopes before consent page
+     *
+     * @param  oAuthMessage oAuthMessage
+     * @param oauth2Params oauth2Params
+     */
+    private void handleAuthorizationBeforeConsent(OAuthMessage oAuthMessage, OAuth2Parameters oauth2Params,
+                                             OAuth2AuthorizeReqDTO authzReqDTO)
+            throws IdentityOAuth2ScopeValidationException, OAuthSystemException {
+
+        try {
+            AuthorizationHandlerManager authzHandlerManager = AuthorizationHandlerManager.getInstance();
+            OAuthAuthzReqMessageContext authzReqMsgCtx = authzHandlerManager.
+                    handleAuthorizationBeforeConsent(authzReqDTO);
+            //add OAuthAuthzReqMessageContext to SessionDataCacheEntry
```

Comment:
Start comments with a capital and end with a full stop.
Check formatting on other places too in the PR

Reply:
Also it would be good if the comment explain why we are doing this action instead of just saying what the action is (this is obvious for this particular case...).

"Why are we setting the context object into this session data cache entry?"